ui/desktop/app/controllers/application.js

```diff
@@ -75,4 +102,34 @@ export default class ApplicationController extends Controller {
         rootEl.classList.remove('rose-theme-light');
     }
   }
+
+  @action
+  signoutAttempt() {
+    this.showSignoutModal = true;
+    // Prevents a user from quitting app while signout modal is present
+    this.ipc.invoke('setSignoutInProgress', true);
+  }
+
+  @action
+  cancelSignout() {
+    this.showSignoutModal = false;
+    this.ipc.invoke('setSignoutInProgress', false);
+  }
+
+  @action
+  closeAppAttempt() {
+    this.showCloseSessionsModal = true;
+  }
+
+  @action
+  confirmCloseSessions() {
```

Comment:
You could just use a shared function here, you just need to check if the app is quitting from the variable your electron handler set to determine whether we should call `this.session.invalidate()` or to close the app.